app/javascript/utilities/calculateTextAreaHeight.js

```diff
@@ -0,0 +1,103 @@
+const HIDDEN_TEXTAREA_STYLE = `
+min-height:0 !important;
+max-height:none !important;
+height:0 !important;
+visibility:hidden !important;
+overflow:hidden !important;
+position:absolute !important;
+z-index:-1000 !important;
+top:0 !important;
+right:0 !important
+`;
+
+const SIZING_STYLE = [
+  'letter-spacing',
+  'line-height',
+  'padding-top',
+  'padding-bottom',
+  'font-family',
+  'font-weight',
+  'font-size',
+  'text-rendering',
+  'text-transform',
+  'width',
+  'text-indent',
+  'padding-left',
+  'padding-right',
+  'border-width',
+  'box-sizing',
+];
+
+let hiddenTextarea;
+
+/**
+ * Helper function to get the height of the textarea based on the current text content
+ *
+ * @param {HTMLElement} uiTextNode The textarea to measure height of
+ *
+ * @returns {{height: number}} Object with the calculated height
+ */
+export const calculateTextAreaHeight = (uiTextNode) => {
```

Comment:
This is a slightly refactored version of the calculation used by the preact-autosize-textarea lib we used to use in the editor body.